CRM/Event/Form/Participant.php

```diff
@@ -524,8 +524,8 @@ public function setDefaultValues(): array {
 
       $this->_eventTypeId = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_Event', $eventID, 'event_type_id', 'id');
 
-      $this->_discountId = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_Participant', $this->_id, 'discount_id');
-      if ($this->_discountId) {
+      if ($this->getDiscountID()) {
+        // This doesn't seem used....
```

Comment:
Agreed that this doesn't seem to be used anywhere. Seems like the previous logic was to get the discount ID from the participant, but as it is here we'd be getting the discount ID from the event itself. Is that always the same? Not sure. Maybe cleaner just to get rid of it, since it does nothing.

Reply:
I wonder if we can just get rid of the the whole getting the discount ID from the participant thing entirely. It doesn't seem to do anything at all. Then we can simplify getDiscountID().

Reply:
@larssandergreen I was referring to the `set` not being used - I think loading it from the participant makes sense in edit mode